src/coreComponents/mesh/generators/CellBlockManager.hpp

```diff
@@ -22,10 +22,88 @@
 #include "mesh/generators/CellBlockManagerABC.hpp"
 #include "mesh/generators/CellBlock.hpp"
 #include "mesh/generators/FaceBlock.hpp"
+#include "mesh/mpiCommunications/SpatialPartition.hpp"
 
 namespace geosx
 {
 
+
+/**
+ * @class PartitionDescriptor
+ * @brief Simple utility to retrieve partition information in case of Metis or Spatial partition.
+ */
+class PartitionDescriptor
+{
+public:
+  /**
+   * @brief indicate if the partition is described using a Metis Neighbor list.
+   */
+  bool hasMetisNeighborList() const { return m_hasMetisNeighborList; }
+  /**
+   * @brief Sets the boolean that indicates if the partition is described using a Metis Neighbor list.
+   */
+  void setHasMetisNeighborList( bool value ) { m_hasMetisNeighborList = value; }
+  /**
+   * @brief Gets a reference to the list of metis neighbor list.
+   */
+  std::set<int> & getMetisNeighborList() {return m_metisNeighborList; }
+  /**
+   * @brief indicate if the partition is described using a spatial partition.
+   */
+  bool hasSpatialParition() const { return !m_hasMetisNeighborList; }
+  /**
+   * @brief Sets the boolean that indicates if the partition is described using a Metis Neighbor list.
+   */
+  void setHasSpatialPartition( bool value ) { m_hasMetisNeighborList = !value; }
+  /**
+   * @brief Returns a reference to the spatialPartition
+   */
+  SpatialPartition & getSpatialPartition() { return m_spatialPartition; }
+private:
+  bool m_hasMetisNeighborList;         //< Indicate if we use metis neighbor list or spatial partition to describe the partition
+  std::set<int> m_metisNeighborList;   //< The list of neighbors computed wwith metis
+  SpatialPartition m_spatialPartition; //< The spatial partition
+};
+
+/**
+ * @struct description of a well block
+ */
+class WellBlock
+{
+public:
+  /**
+   * @brief Getter for the well name
+   */
+  const string & getName() { return m_name; }
+  /**
+   * @brief Getter for the well region name
+   */
+  const string & getWellRegionName() { return m_wellRegionName; }
+  /**
+   * @brief Getter for the well controls name
+   */
+  const string & getWellControlsName() { return m_wellControlsName; }
```

Comment:
I'm surprised that this is the only information that's being created. I'd expect data like
```c++
  /// Physical location of the nodes
  array2d< real64 > m_nodeCoords;

  /// Global number of perforations
  globalIndex m_numPerforations;

  /// Absolute physical location of the perforation
  array2d< real64 > m_perfCoords;

  /// Well Peaceman index at the perforation
  array1d< real64 > m_perfTransmissibility;

  /// Global index of the well element
  array1d< globalIndex > m_perfElemId;
```
to appear there? Maybe it's WIP?